gensim/corpora/svmlightcorpus.py

```diff
@@ -111,6 +112,8 @@ def save_corpus(fname, corpus, id2word=None, labels=False, metadata=False):
         """
         logger.info("converting corpus to SVMlight format: %s", fname)
 
+        if isinstance(labels, np.ndarray):
+            labels = labels.tolist()
```

Comment:
Needs a comment: np arrays are sequences too, so it's not obvious why they should be a special case.

By the way, should they? Can't we check for "emptiness" in a way that works for arbitrary sequence, without singling out numpy arrays?

Reply:
@piskvorky if we simply do,
```
if  labels:
    labels = list(labels)
```
it would surely work for arbitrary sequences, so should the documentation also be changed to reflect that support?

Reply:
I don't think the `if` will work -- won't it have the same problem as the current version, if `labels` is a numpy array?

Reply:
@piskvorky  would this be good enough to use?? seems to working fine 
```
try:
    labels = list(labels)
except:
    pass
```

Reply:
What the `except` for?

Reply:
@piskvorky `labels` can be `False`/`None` (in case if no labels), maybe for this we really need to use `isinstance`.

Reply:
Aha, I see. Then the try/catch makes sense (but needs to catch a specific Exception, and of course a comment explaining what's happening).

Reply:
@piskvorky would this be good for a commit
```
# If arbitrary sequences (e.g numpy array) is given as input to labels, casting
# to list will generate TypeError if labels is False (empty)
try:
    labels = list(labels)
except TypeError:
    pass
```

Reply:
I don't really understand that nested `if` in the comment. Why should casting a numpy array to list generate a TypeError?

Reply:
@piskvorky because `labels` can be "empty" (see my previous comment https://github.com/RaRe-Technologies/gensim/pull/2197#discussion_r220801693)

See an examples

```python
>>> list(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```

```python
>>> list(False)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bool' object is not iterable
```

Reply:
I understand how TypeError is thrown when `labels` is `False`. But @aquatiko 's code comment implies an exception in case `labels` is a numpy array. But maybe it's just badly phrased?

Reply:
@piskvorky will this comment work?
```
# When labels=False, the operation for casting arbitrary sequences (e.g numpy array)
# to list will cause a TypeError 
```

Reply:
@piskvorky  This looks better than the try/except method. Do the docstring needs to be changed? As ```labels``` will support arbitrary sequences now and its written-- labels : list or False

Reply:
I think so, yes. @menshikh-iv do we write `sequence` or `iterable` in such cases?

Reply:
@piskvorky `iterable of TYPE`